src/parse/asp/grammar.go

```diff
@@ -42,7 +42,9 @@ type FuncDef struct {
 	// Not part of the grammar. Used to indicate internal targets that can only
 	// be called using keyword arguments.
 	KeywordsOnly bool
-	IsPrivate    bool
+	// allowed return type of the FuncDef
+	Return    string
```

Comment:
super nit, please move `Return` before `KeywordsOnly` in the struct; the current layout is inefficient (there is a Go linter for this but we don't use it during CI or anything)